common/persistence/client/health_request_rate_limiter.go

```diff
@@ -149,19 +157,36 @@ func (rl *HealthRequestRateLimiterImpl) maybeRefresh() {
 }
 
 func (rl *HealthRequestRateLimiterImpl) refreshRate() {
```

Comment:
I'm surprised this is an issue at all, the refresh methods are only called behind a read from the timer channel which I thought was thread-safe in go. Are you sure it's not an issue with our test setup?

Reply:
Maybe L109 in the constructor just needs to be changed from `limiter.refreshDynamicParams()` to `limiter.maybeRefresh()`. That's the only place refresh is called without going through the timer.

Reply:
To be clear, I didn't see data race error in this function, just thought that this is called right after `refreshDynamicParams` (which is where I saw the issue), so it could have the same issue.